src/jit/lsraxarch.cpp

```diff
@@ -2342,7 +2342,9 @@ void LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
         case NI_SSE2_CompareNotEqualOrderedScalar:
         case NI_SSE2_CompareNotEqualUnorderedScalar:
             info->internalIntCount = 1;
+#ifdef _TARGET_X86_
             info->setInternalCandidates(this, RBM_BYTE_REGS);
+#endif
```

Comment:
Only 32-bit platforms have this limitation.

Reply:
Could you please elaborate. The tmpReg is used by `sete`, which requires it to be a byte register.

Reply:
On x64 all registers are byte registers so the call is useless. That said, I think I prefer the existing code, `RBM_BYTE_REGS` is `RBM_ALL_INT` on x64 so the code is correct and one less `ifdef` is a good thing.

Reply:
>  I think I prefer the existing code

I agree.

Reply:
@mikedn I see, `RBM_BYTE_REGS` already deal with the platform difference. Thank you for pointing out.